Public/Src/Cache/ContentStore/Vsts/DedupReadOnlyContentSession.cs

```diff
@@ -155,14 +172,55 @@ public class DedupReadOnlyContentSession : ContentSessionBase
                 return new PinResult($"DedupStore client requires {RequiredHashType}. Cannot take HashType '{contentHash.HashType}'.");
             }
 
-            try
+            var pinResult = CheckPinInMemory(contentHash);
+            if (pinResult.Succeeded)
             {
-                var pinResult = CheckPinInMemory(contentHash);
-                if (pinResult.Succeeded)
-                {
-                    return pinResult;
-                }
+                return pinResult;
+            }
+
+            VstsBlobIdentifier blobId = ToVstsBlobIdentifier(contentHash.ToBlobIdentifier());

```

Comment:
Actually, this is the place for a paragraph of comments describing the basic approach.